### DISCUSSION TITLE
还是同样的报错，网络请求返回: {"data":null,"message":"","retcode":1028}，但是是星铁的报错

### DISCUSSION THREAD

**Discussion by pbxs666** [Ljzd-PRO/Mys_Goods_Tool#137] [Q&A]
如标题，很多朋友发帖说了这个问题，米肯定是把什么东西给改了，但是波及的不仅仅是**原神**，**星穹铁道**也是这样

06-16 18:00:00 [INFO] mys_goods_tool.api | 米游币商品兑换 - 执行兑换: 用户 136508784 商品 2023052611696 兑换失败，可以自行确认。
06-16 18:00:00 [DEBUG] mys_goods_tool.api | 网络请求返回: {"data":null,"message":"","retcode":1028}
06-16 18:00:00 [ERROR] mys_goods_tool.exchange_mode | 用户 136508784 - 【崩坏：星穹铁道】鼠标垫-星旅一瞬 - 线程 1 - 兑换失败
06-16 18:00:00 [INFO] mys_goods_tool.api | 米游币商品兑换 - 执行兑换: 用户 136508784 商品 2023052611696 兑换失败，可以自行确认。
06-16 18:00:00 [DEBUG] mys_goods_tool.api | 网络请求返回: {"data":null,"message":"","retcode":1028}
06-16 18:00:00 [ERROR] mys_goods_tool.exchange_mode | 用户 136508784 - 【崩坏：星穹铁道】鼠标垫-星旅一瞬 - 线程 2 - 兑换失败
06-16 18:00:00 [INFO] mys_goods_tool.api | 米游币商品兑换 - 执行兑换: 用户 136508784 商品 2023052611696 兑换失败，可以自行确认。
06-16 18:00:00 [DEBUG] mys_goods_tool.api | 网络请求返回: {"data":null,"message":"","retcode":1028}
06-16 18:00:00 [ERROR] mys_goods_tool.exchange_mode | 用户 136508784 - 【崩坏：星穹铁道】鼠标垫-星旅一瞬 - 线程 3 - 兑换失败

PS：使用最新版2.0.4

**Comment by Aellyt** [comment]
之前是只有原神，怀疑米在尝试性的覆盖过去

**Comment by Ljzd-PRO** [comment]
米哈游API的域名从挺早之前开始就改成了 `miyoushe.com`，但应该是为了兼容旧版本App，所以原先的API也可以用。现在过去了数个版本，可能是打算放弃旧版本API支持了。但域名以外别的可能变动较少。

**Comment by kuronekoda** [comment]
请问你有没有解决，我和你一样星铁兑换失败
06-30 17:58:34 [INFO] mys_goods_tool.exchange_mode | 已添加定时兑换任务 74510315 - 【崩坏：星穹铁道】鼠标垫-星旅一瞬 - 2023-06-30 18:00:00
06-30 18:00:00 [INFO] mys_goods_tool.api | 米游币商品兑换 - 执行兑换: 用户 74510315 商品 2023052611696 兑换失败，可以自行确认。
06-30 18:00:00 [DEBUG] mys_goods_tool.api | 网络请求返回: {"data":null,"message":"","retcode":1028}
06-30 18:00:00 [ERROR] mys_goods_tool.exchange_mode | 用户 74510315 - 【崩坏：星穹铁道】鼠标垫-星旅一瞬 - 线程 1 - 兑换失败
06-30 18:00:00 [INFO] mys_goods_tool.api | 米游币商品兑换 - 执行兑换: 用户 74510315 商品 2023052611696 兑换失败，可以自行确认。
06-30 18:00:00 [DEBUG] mys_goods_tool.api | 网络请求返回: {"data":null,"message":"","retcode":1028}
06-30 18:00:00 [ERROR] mys_goods_tool.exchange_mode | 用户 74510315 - 【崩坏：星穹铁道】鼠标垫-星旅一瞬 - 线程 2 - 兑换失败

**Comment by Ljzd-PRO** [comment]
又尝试修复了一下，可以试一下：https://github.com/Ljzd-PRO/Mys_Goods_Tool/actions/runs/5479310149

解决了几个可能的问题：
- `device_config`, `salt_config` 如果程序默认值更新了但是用户数据文件用的还是旧的，旧的配置还是会覆盖掉默认值的问题
- 兑换请求的Headers里增加`Referer`，修改`Origin`, `Content-Type`
- Headers的`User-Agent`里的App版本号部分和`x-rpc-app_version`不一致的问题

除了Headers里没有`x-rpc-verify_key`, `x-rpc-device_fp`，其他应该都和米游社App里实际发出的请求一致了。
